File: Arrays & String/Problem1 - 1768.java
public static String mergeAlternately(String word1, String word2) {
    int n = word1.length();
    int m = word2.length();
    StringBuilder mergeWord = new StringBuilder();
    int i = 0, j = 0;
    while (i < n || j < m) {
        if (i < n) mergeWord.append(word1.charAt(i++));
        if (j < m) mergeWord.append(word2.charAt(j++));
    }
    return mergeWord.toString();
    /*  TC-O(max(n, m)) SC-O(1) */
}
 

File: Arrays & String/Problem1 - 1768.cpp
#include <bits/stdc++.h>
using namespace std;

/* Optimal Approach*/
string mergeAlternately(string word1, string word2) {
	int n = word1.size(), m = word2.size();
	string mergeWord = "";
	int i = 0, j = 0;
	while (i < n || j < m) {
		if (i < n) mergeWord += word1[i++];
		if (j < m) mergeWord += word2[j++];
	}
	return mergeWord;

	/* TC-O(max(n, m)) SC-O(max(n, m)) */
}
